shared/p2p/feed.go

```diff
@@ -35,8 +35,10 @@ func (s *Server) Feed(msg interface{}) *event.Feed {
 		t = reflect.TypeOf(msg)
 	}
 
+	s.mutex.Lock()
```

Comment:
Please add `defer s.mutex.Unlock()` on the next line and remove the other unlock. This makes it more clear that you’re releasing the locked resource.

Reply:
Sure, done.